libp2p/stream/lpstream.nim

```diff
@@ -304,9 +340,9 @@ proc closeWithEOF*(s: LPStream): Future[void] {.async, public.} =
     var buf: array[8, byte]
     if (await readOnce(s, addr buf[0], buf.len)) != 0:
       debug "Unexpected bytes while waiting for EOF", s
+  except CancelledError:
```

Comment:
why has this been changed?

Reply:
for `CancelledError`: the callers just end up discarding it. discarding it earlier allows `{.async: (raises: []).}`

for `CatchableError`, it cannot be raised by `readOnce`, catching `LPStreamError` is enough, and the compiler guarantees exhaustiveness because the proc itself is tagged as `{.async: (raises: [].}`, so it would alert us if an exception is forget to be caught.